### ISSUE TITLE
Chaining together multiple responses

### ISSUE THREAD

**Issue by millerj-tc** [ACG8/Lucid#24]
What if player A activates **Alaea's** ability from the new filler set:
> Barrier 5: You gain 3 points and may cue the diver’s bottom three cards in any order. If you do, you may immediately dive them three times, ignoring traps.

Then after the first access, someone plays **Kaen** from Mercenary:
>Lucid T: After another player accesses a card that isn’t their own, access that player’s top card.

Which in turn reveals a **Memory Trader** from Therapist:
>Trap: The diver gains 1 point. You may immediately dive their mind with depth+2.

Now there's a million things happening. It's not necessarily bad in this case.  I think it just goes:

**Alaea dive 1** -- **Kaen access** -- **Memory Trader dive** -- **Alaea dive 2** -- **Alaea dive 3**

It also is awkward because what if **Memory Trader** was another card that let the trap owner dive anyone. They dive the **Alaea** owner who is already in the process of being dived with cards revealed by the first player's dive. So would the alternate **Memory Trader** dive start with the first card under **Alaea**?

Maybe we should rewrite these to saying something like "ready a dive" rather than "immediately dive" and then you have to resolve each dive/access before the next one can start?

**Comment by ACG8** [comment]
Yes, this is a bit of a problem. Putting aside the issue of interrupting a dive to perform another dive, having to keep track of multiple dives is a nuisance.

One way to resolve this would be to make dives a resource, and to go around in turn order until nobody has any dives left. However, this prevents non-plunge dives (or makes the rules significantly less elegant).

I don't see a problem with somebody else diving one of the cued cards from Alaea; if anything, it is a nice bit of counterplay. The main issue is resolving all of the effects and keeping track of who is diving in what order.

I don't think saying "ready a dive" really solves the problem; there can still be way too many things to keep track of. Interrupting an access to dive is annoying, but not game-breaking. We just need to be a little careful in how we word accesses (e.g. accessing the top card of the mind is not an issue, but accessing a specific card whose position you may not know after a dive could be problematic).

I'm not seeing a good way to simplify things at the moment without restricting design space by an unacceptable amount.

**Comment by millerj-tc** [comment]
How much does it help if Alaea and Memory Trader are worded as "...dive...after the current dive is resolved?"

**Comment by millerj-tc** [comment]
It means that at least you don't wind up in the situation where you are diving someone who still has cards revealed by depth because they are in the middle of being dived.

**Comment by millerj-tc** [comment]
# Top Priority
Keep the base game rules simple enough that I can play with my parents -- not solving this by adding a ton of complexity to the simplest version of the game.

Part of the problem is that if you channel **Mexxos** (Thieves) off of a plunge in another player's mind and then decide to dive the same player you still run into the problem of cards revealed through depth being removed when the "card dive" happens. We will probably need to combine some of these solutions with the "after the current dive is resolved" wording.

**Mexxos** would read something like:

>Channel 4: After the dive resolves, the diver plunges into another player’s mind with depth +4. They ignore the first barrier ability they trigger.

# My preferred fix
The main problem seems to be with dives that reveal multiple cards and get interrupted? Like chaining together 5 **Trinket Find** (Thieves) channels is easy to keep track of. Maybe we design cards like this:

* A single access/basic dive is always fine (they will always resolve one at a time)
* Multi-reveal dives/plunges _always_ ignore dive initiation effects (even if they just have the capability to be multi-reveal -- any cards that allow a plunge even if they don't give a depth modifier in the effect)
* Lucid card dives are always worded "after a dive resolves" that way it still happens one at a time

Channeled dives would need to be designed such that they never give anyone other than the diver a "card dive."  Nothing like:

> Channel: Gain 1 point.
> When this is channeled, the owner may dive another player 

This would need to be combined with the example **Mexxos** fix from the first section.

>Channel 4: After the dive resolves, the diver plunges into another player’s mind with depth +4. Ignore dive initiation effects.

**Memory Trader** would read:

> Trap: The diver gains 1 point. After the dive resolves, you may dive their mind with depth 2. Ignore dive initiation effects.

**Alaea** would read:

> Barrier 5: You gain 3 points and may cue the diver’s bottom three cards in any order. If you do, you may immediately dive them three times. Ignore traps and dive initiation effects.

This would stop you from chaining **Mexxos** into **Mexxos**, but maybe that's okay?

We could also make this a rulebook fix rather than a card text fix.

We say in the rules: "A card that instructs you to dive another player before the current dive has been resolved is ignored." The only weakness of this is that it means that multi-dives/accesses like **Alaea's** will always be off-limits.

We could even cut one word off of the card text if we changed the "dive" card type to something else (it's already weirdly redundant.)

**Memory Trader** could read:

> Trap: The diver gains 1 point. After the dive resolves, you may dive their mind with depth 2. Ignore dive effects.

This still leaves the **Kaen**/**Alaea** loophole, since **Kaen** references accesses, but we can reword **Kaen** or nerf **Alaea** to a single access or change her to a dive with depth 2 (since multiple accesses don't appear anywhere else yet. We could make that a hard design limit to help with this problem.) Although it should probably just be **Kaen** because maybe it's safer to just not let people respond to an access and limit the responses to resolved dives to avoid these kinds of issues.

I like this fix because it shouldn't make playing with my parents any harder either way. If you just play the party set (the simplest version of the game), you won't be playing with these cards anyway. So either they won't have to parse the card text ("ignore dive initiation effects") or I can just skip over that rule when I'm explaining the game.

# Another Rulebook fix
We could go back to distinguishing "card dives" and they could put the card in front of them (we could put this in the rules rather than on the card). Then you resolve any "card dives" clockwise from FPT and if more pop up you keep going round and round. At least there's a way to keep track of them then.

# Dive Zone fix
We could give everyone a generic "dive" card that describes dives and plunges. Almost like a rules reminder.  You put it in your "dive zone." You flip it over when you use it. BUT we write out **Memory Trader** like this:

> Trap: The diver gains 1 point. Put this card in your dive zone.
> Dive: Dive the player who activated the trap with depth 2.

But you can only activate cards in the dive zone in player order when no other dive is happening. You still have to remember who triggered it which is more bookkeeping. This also adds more complexity to the basic mechanics (dive zone). I mostly hate this idea, but I already wrote it, so I'm leaving it for posterity.

# Design space fix
We could limit card dives to one card type -- maybe Lucid cards? That way you can't chain them together like you potentially could with anything else. Or maybe channel cards but making sure that there's no way for an opponent to interrupt an active "card dive."

# Removal fix
I really like the design space of cards having dive effects, but we may also want to step back and evaluate how much it's really adding. It may be something we can do without.

**Comment by ACG8** [comment]
I don't like Dive Zone Fix.

One variant on Design Space Fix that I think could work well is to make sure that non-plunge dives are only triggered off of plunges (not off of other non-plunge dives). So you never have an effect that allows you to dive in response to a dive or access that is not part of a plunge. Since each player gets 1 plunge per turn (and we can control additional plunges if we add them later), this prevents long dive-counterdive chains.

**Alaea**

> Barrier 5: Only triggers during a plunge. You gain 3 points and may cue the diver’s bottom three cards in any order. If you do, you may immediately dive them three times, ignoring traps.

**Mexxos**

> Channel 4: Only usable during a plunge. The diver may immediately dive another player’s mind with depth +4. They ignore the first barrier ability they trigger. (This card may not be channeled again this phase.)

**Memory Trader**

> Trap: Only triggers during a plunge. The diver gains 1 point. After the end of this plunge, you may dive their mind with depth 2. Ignore dive effects.


Solutions such as Another Rulebook Fix and Dive Zone Fix that involve moving the cards providing the extra dives are problematic because in practical terms we would have to forget where the cards came from, which prevents us from including dive effects on cards that are meant to stay in place when triggered (barriers, in particular).

The Preferred Fix would prevent us from designing dive barriers (such as Memory Trader). Not to say we shouldn't do it, but it would be sad to lose barriers of that sort.

**Comment by millerj-tc** [comment]
I'm not sure what you mean by "dive barriers" (**Memory Trader** is a trap?)

I outlined how it would work in the preferred fix section and it works almost identically to the way it does now except it can't trigger another dive effect and waits until the current dive is resolved. I guess one last difference is that if you are using a card dive, **Memory Trader** no longer triggers. I don't see that as a huge loss since there aren't that many card dive effects and none of them are repeatable, so at most that might come up once per game?

Also only-triggers-off-plunges (OTOP) solution means we can't design cards that allow plunges (spending additional resources), only fixed depth dives. (The **Mexxos** effect is a plunge.) So it seems like it's kind of a wash in terms of design space. Either way, we're limiting something.

I think it's cleaner to say that card dives never trigger other card dives. Every card we've designed still works the way it should in any situation other than when it is revealed/accessed during a card dive which is already an infrequent scenario. I think we should preference  opening general design space at the expense of closing design space for an already infrequent scenario. The OTOP solution does it the other way: it keeps the (narrower) design space of counter-diving off of a card dive to  close the general design space of designing card dives that allow plunges.

**Comment by millerj-tc** [comment]
The OTOP solution is also awkward because how do you handle "plunge only" channels on the top of your own mind. Can you plunge for 0 resources? If you can that makes designing cards that care about plunging weird because it *can* be effecticely the same as a dive but technically a plunge by paying 0.

**Comment by ACG8** [comment]
> I'm not sure what you mean by "dive barriers" (Memory Trader is a trap?)

You're right; a brief and inexplicable memory lapse on my part. We may never know what was going through my head when I wrote that.

> Also only-triggers-off-plunges (OTOP) solution means we can't design cards that allow plunges (spending additional resources), only fixed depth dives. (The Mexxos effect is a plunge.) So it seems like it's kind of a wash in terms of design space. Either way, we're limiting something.

Is it important to allow players to plunge multiple times per turn, though? This would allow players to access multiple secrets in the same round; not sure if that is a good thing or not.

That said, I am not opposed to cards that allow plunges, just to cards that allow plunges as a reaction to a dive. If you have a card that gives you an additional plunge (but not as a reaction to another player's dive or access), OTOP does not forbid that.

My proposed rewrite of Mexxos still provides the player with a decently strong dive, just not one that can be turned into a run on another player's secret.

> I think it's cleaner to say that card dives never trigger other card dives. Every card we've designed still works the way it should in any situation other than when it is revealed/accessed during a card dive which is already an infrequent scenario.

The one thing I don't like about this solution is that it requires drawing a distinction between dives based on their source. If plunges are a special dive permitted once per turn (or even multiple times per turn from cards but never as a reaction), there is a natural distinction between a plunge and a generic dive. The Preferred Method (TPM) requires creating a second rules distinction based on the source of the dive. And it means that if a card provides a player with an extra plunge (not as a reaction) it would not be treated the same as the plunge the player gets to make as a part of their turn. I would rather restrict the scenarios in which you can gain a plunge and treat all plunges the same, regardless of source.

**Comment by ACG8** [comment]
> The OTOP solution is also awkward because how do you handle "plunge only" channels on the top of your own mind. Can you plunge for 0 resources? If you can that makes designing cards that care about plunging weird because it can be effectively the same as a dive but technically a plunge by paying 0.

I have no objection to depth+0 dives. All plunges are dives, so the difference between a depth+0 plunge and a depth+0 dive is the same as the difference between a depth+3 plunge and a depth+3 dive.

**Comment by ACG8** [comment]
Here is how I would reword the rulebook under OTOP (absent some refinements to my wording), or even just generally to clarify the distinction between plunges and dives:

...

## Dive phase
During the Dive phase, each player in turn order may execute a **plunge**. A **plunge** is a special type of dive in which a player may optionally pay resources to increase the depth of the dive. To plunge, perform the following steps:
* Declare which mind you want to dive into. You may dive into your own mind or another player’s mind.
* Declare how many resources you want to pay to increase the depth of the plunge. Each resource spent increases the depth of the plunge by 1. Set aside all resources paid; some card effects may cause them to be refunded.
* Resolve a dive against the target player's mind with the modified depth.
### Resolving a dive
When you resolve any dive against a player (plunge or otherwise), perform the following steps.
* Reveal as many cards from the top of the mind one at a time as your _depth_ modifier. 
  * If you reveal a card with a barrier ability in another player’s mind you must resolve its effect before continuing your dive or accessing that card (barriers in your own mind do not affect your dive). This is called "triggering" the barrier. 
  * Set aside cards you dove through face up.
  * You may stop your dive at any time to skip to step 3 and access the card currently revealed.
* Reveal the top card of the mind deck, this is called the access. If the card is from another player’s mind, you may cue it or bury it. If the card is from your mind, you may also draw that card.
  * If you access a phantasm card, lose 1 point.
  * If you access a card with a channel ability, you may use it. After the effect is resolved, put the card on the bottom if its mind. A card can only be channeled once per phase.
  * If you access a card with a trap ability, that ability activates. After the effect resolves, put the card on the bottom if its mind. (When a trap card refers to “you” it is referring to the controller of the card.)
  * If you are diving another player’s mind and they have no cards remaining in their mind (because they have been set aside during your dive because of depth), you may look at that player’s secret card (don’t reveal it to other players). You may expose that card, using its expose ability immediately. If you do, turn that card face up. That card cannot be exposed again this game.

> The accessed card is still on top of its mind.

> When a channel card says something like "the diver may choose another player" -- they may choose the player whose mind they dived, just not themselves.

* Shuffle any cards revealed with your depth modifier and return them to the top of their mind.
* Pay any non-refunded resources to the bank.
* The dive is now resolved.

...

**Comment by millerj-tc** [comment]
Clarifications:

>Is it important to allow players to plunge multiple times per turn, though? This would allow players to access multiple secrets in the same round; not sure if that is a good thing or not.

Not sure how you mean. If I channel **Mexxos** I already used my regular dive to channel it. If I use the **Mexxos** plunge to access another **Mexxos** then it's true that I can plunge again, but I can't go:

Regular dive->Channeled Dive 1->Secret->Channeled Dive 2 -> Secret

Unless Channeled Dive 1 allows multiple dives/plunges which seems like a bad idea anyway. (I'm going to change **Alaea** to be a dive with depth 2 instead of a triple dive.)

OTOP actually stops this line of play. What if I Mexxos into Mexxos and don't want to channel anything else? I just wasted my turn. TPF allows you to Mexxos into Mexxos if that is your only/best line of play.

I guess you could plunge for your turn and then counter-plunge off of a barrier activation? We could limit plunges to only channel cards pretty easily.

> The one thing I don't like about this solution is that it requires drawing a distinction between dives based on their source.

But OTOP requires creating a subset of channels, traps, and barriers that can only be accessed on plunges, so that's just creating three new distinctions somewhere else in the game.

Also, since TPF prints the rules right on the card (like OTOP), it's not like it's something they have to reference in the rulebook. It's right there for them to see.

------

It's pretty close. With either OTOP or TPF we need to add rules text to the card. When it comes down to it, it's the difference between:

* Plunges are a special kind of dive and some cards are only triggered by plunges, which we communicate by writing on the card "Only triggered on a plunge"
* Card dives are a special kind of dive, and some cards can't be triggered by card dives, which we communicate by writing on the card "Ignore dive effects."

We're just saying the same thing opposite ways right?

------

TPF solution requires the fewest characters:

"Only triggers during a plunge."
vs
"Ignore dive effects."

This solution also is the most intuitive in my opinion. I think the OTOP text would feel weird and arbitrary, like "Why can certain cards only be channeled on plunges?" TPF addresses the root problem in the most straightforward way.

------

TPF advantages:
- lines of play (Mexxos->Mexxos is legal, but still easy to keep track of because of "after dive is resolved terminology")
- design space (allows plunges off of channels which OTOP forbids)
- fewer characters on the card

disadvantages:
- cares about the source of a dive

I think the disadvantage is mitigated pretty thoroughly by the fact that even though the rules "care" about dive sources in an abstract sense, the practical implementation is putting the rules text on a card, which OTOP does anyway. We don't need to change the rulebook at all for TPF to work. Not that changing the rulebook is a big deal, I just mean that the fact that the rules "care" doesn't add complexity except in the exact situation where this will come up and then the card tells you exactly how to handle it.

**Comment by ACG8** [comment]
All right, let's go with TPF then and see how it works. I'll see if any of my cards need to be changed for this.
